### PR TITLE
test(dtslint): add tap operators dtslint test (#4093)

### PR DESCRIPTION
**Pull request by dkosasih**
**Description:** Add dtslint test for tap operators

**Related issue (if exists):** #4093

### CODE REVIEW

File: spec-dtslint/operators/tap-spec.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { of } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const a = of(1, 2, 3).pipe(tap()); // $ExpectType Observable<number>
+});
+
+it('should accept partial observer', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: number) => x })); // $ExpectType Observable<number>
+  const b = of(1, 2, 3).pipe(tap({ error: (x: number) => x })); // $ExpectType Observable<number>
```
Comment (cartant) [inline spec-dtslint/operators/tap-spec.ts]: The `error` callbacks should be typed as `any`. There is no guarantee that the callback will be passed a particular type. Also, the error callback should not return a value. Use `error: (err: any) => {}`.

File: spec-dtslint/operators/tap-spec.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  const c = of(1, 2, 3).pipe(tap({ complete: () => 1 })); // $ExpectType Observable<number>
+});
+
+it('should enforce type for next observer function', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: string) => x })); // $ExpectError
+});
+
+it('should accept any type for error observer function', () => {
+  const a = of(1, 2, 3).pipe(tap({ error: (x: string) => x })); // $ExpectType Observable<number>
```
Comment (cartant) [inline spec-dtslint/operators/tap-spec.ts]: Use `error: (err: any) => {}`. The error callback should accept `any` and should not return. TypeScript is okay with it, but the test should reflect the intended usage.

File: spec-dtslint/operators/tap-spec.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { of } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const a = of(1, 2, 3).pipe(tap()); // $ExpectType Observable<number>
+});
+
+it('should accept partial observer', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: number) => x })); // $ExpectType Observable<number>
```
Comment (cartant) [inline spec-dtslint/operators/tap-spec.ts]: Don't return a value from the `next` handler. TypeScript is okay with this, but the test should reflect the intended usage.

**Comment by dkosasih** [comment]
@cartant 
I did the changes. Could you have a look, please?

File: spec-dtslint/operators/tap-spec.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+it('should not accept empty observer', () => {
+  const a = of(1, 2, 3).pipe(tap({})); // $ExpectError
+});
+
+it('should enforce type for next observer function', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: string) => { })); // $ExpectError
+});
+
+it('should accept any type for error observer function', () => {
```
Comment (cartant) [inline spec-dtslint/operators/tap-spec.ts]: This test can be removed, as it's the same as the one in the `should accept partial observer` test.

**Pull request merged**